### DISCUSSION TITLE
Notifications/Toast Component

### DISCUSSION THREAD

**Discussion by divine** [tailwindlabs/headlessui#439] [Ideas]
Notifications as a component would be a great addition.

https://tailwindui.com/components/application-ui/overlays/notifications

https://ant.design/components/notification/
https://vue.chakra-ui.com/toast

**Comment by holtergram** [comment]
I like these examples a lot
[Vuesax Notifications](https://vuesax.com/docs/components/Notification.html#sticky)

**Reply by RejownAhmed** [reply]
Try this: However, I did not include tailwind yet, although examples are with tailwind. https://vue-modern-notification.netlify.app/

**Reply by RejownAhmed** [reply]
Any contribution is welcome. Also, it would be great if a notification component is introduced in Headless UI

**Reply by instanceasia** [reply]
I’ve been experimenting with a small toast utility that focuses on:
- minimal bundle size
- simple API
- stacked notifications similar to those examples

If you're curious about the approach, it's here:
[vue-toast-tailwind](https://www.npmjs.com/package/vue-toast-tailwind)

**Comment by mquadrat** [comment]
This is really missing as it is one of the components in TailwindUI. And quite frankly the existing notification plugins (at least for vue) mostly suck.

**Comment by binajmen** [comment]
Up 🆙 That would be awesome 👍

**Comment by james-gardner** [comment]
Isn't a toast message just another form of modal dialogue? In which case what's lacking here is the behaviour behind managing their display and disposal. I'm not sure that has a place in this library. Open to being convinced otherwise.

**Reply by mquadrat** [reply]
Well, that might be an option. However it would require a lot of custom code (or I'm just thinking too complicated right now). I'll see if I can come up with a solution that uses modals.

**Reply by jstnmthw** [reply]
I stumbled upon this thread for this very thing. I was looking for a way to remove the Dialogs default of adding padding and removing scroll overflow because this is unessecary for the toast notification. As of now, I can't find a solution. I have everything working intended for Dialog to be a Toast notfication except those stylings being added.

**Comment by DannyDelott** [comment]
For anyone still curious about this, we were able to use react-hot-toast with the Tailwind UI notification components pretty easily.
https://react-hot-toast.com/docs/version-2

**Reply by timolins** [reply]
Make sure to check [this example](https://codesandbox.io/s/qvjri) on how to combine headless-ui with react-hot-toast. 👍

**Reply by 4thel00z** [reply]
this crashed my tab

**Comment by williamlmao** [comment]
+1 for toasts. Modal doesn't work because it overlays on top of everything and prevents interaction with other elements on the page.

**Reply by Harrald** [reply]
+1. Modal is not suitable for this.

**Reply by fxmkdev** [reply]
I also just tried with a Modal component until I realized the scroll locking (it's even mentioned in the docs). A Toast component would be nice.

**Reply by Harrald** [reply]
A modal also steals focus and makes the rest of the page inert. This should not happen with a toast message.

**Comment by lennartzellmer** [comment]
+1
Something like radix toast would be neat
https://www.radix-ui.com/docs/primitives/components/toast

**Comment by arthtyagi** [comment]
I've been trying to incorporate Transitions (used in TailwindUI examples) and lol, they simply don't work. The state `open` is always stuck after the first render unless I manage to do a force update.

**Comment by html-3** [comment]
Any updates about Dialogs scroll locking?

**Comment by rizen** [comment]
I'd also love this

**Comment by claide** [comment]
I need this one too

**Comment by JameelKhan9** [comment]
+1

**Comment by ereburg** [comment]
Someday, I believe..)

**Comment by ElliottJPerry** [comment]
🍞+🔥 🙏

**Comment by claudiu909** [comment]
+1

**Comment by swa-desmat** [comment]
+1

**Comment by meotimdihia** [comment]
Sorry for being rude, but development is super slow, we just get 1 new component/year.
Do we have to switch to [react-aria](https://react-spectrum.adobe.com/react-aria/getting-started.html).

And even Tailwindcss founder is using it too:
[![image](https://github.com/tailwindlabs/headlessui/assets/300961/e35f4443-1022-41e7-9945-02ad63f052dd)](https://youtu.be/CLkxRnRQtDE?t=5349)

**Reply by mnlfischer** [reply]
I think the biggest issue is the lack of communication from maintainers. They force you to start discussion in their contribution guide and don't communicate. If also fine if they reject ideas to make clear what is the goal but saying nothing is not nice.

**Reply by meotimdihia** [reply]
@mnlfischer TailwindCss founder confirmed we should use other plugins like `react-aria` on Twitter.
This plugin was built because other plugins don't support Vue. While they are still maintaining this plugin, they don't have a team dedicated to developing new features.

**Reply by mnlfischer** [reply]
@meotimdihia thanks for this info! Do you have any referenced twitter post?

**Reply by meotimdihia** [reply]
@mnlfischer He posted here: https://twitter.com/adamwathan/status/1719119230518374478

**Reply by mnlfischer** [reply]
@meotimdihia thanks for pointing me to it!

**Reply by meotimdihia** [reply]
@mnlfischer they released Catalyst: https://catalyst.tailwindui.com/docs
It used headless UI too.

**Comment by jayatulsiani** [comment]
yess, need this for vue

**Comment by sebastiantheobald** [comment]
+1

**Comment by tnovau** [comment]
👍🏼
